src/backend/distributed/operations/stage_protocol.c

```diff
@@ -411,6 +411,14 @@ CreateAppendDistributedShardPlacements(Oid relationId, int64 shardId,
 	{
 		int workerNodeIndex = attemptNumber % workerNodeCount;
 		WorkerNode *workerNode = (WorkerNode *) list_nth(workerNodeList, workerNodeIndex);
+
+		if (NodeIsCoordinator(workerNode))
+		{
+			ereport(NOTICE, (errmsg(
```

Comment:
if it's never allowed, could we maybe exclude it from the list in master_create_empty_shard? (i.e. ignore shouldhaveshards)

Reply:
It is actually never allowed, and what you mentioned was the first draft, commits have disappeared because of my force push :)

A new variant of `DistributedTablePlacementNodeList`, for example `AppendDistributedTablePlacementNodeList` can be done. However, more checks are needed in the shard placement policies other than round robin, because they don't use the list from `DistributedTablePlacementNodeList`. https://github.com/citusdata/citus/blob/master/src/backend/distributed/operations/stage_protocol.c#L185

With the current solution, there are minimal changes and also it can be understood clearly if not enough shard replicas were created because of prevention of empty placement in the coordinator.

Reply:
@marcocitus @halilozanakgul my first solution is not shown here because of force push but you can check from https://github.com/citusdata/citus/tree/issue4237-2. It probably needs improvement but it applies the idea of not putting the coordinator in any list beforehand. As I said, in my solution I have to take care of different shard_placement policies (maybe I am missing something ?). On the other hand, checking it afterwards like @onderkalaci suggested is brief and quite readable. 

@halilozanakgul in that solution I am also giving a hint of reducing shard_replication_factor.

I would appreciate your opinions to decide which approach is best for this. :)

Reply:
> With the current solution, there are minimal changes and also it can be understood clearly if not enough shard replicas were created because of prevention of empty placement in the coordinator.

ok, that's reasonable. 

Could you:
a) Add a test that is successful (e.g. replication factor 2)
b) check the group IDs of the shard in pg_dist_placement in the test
b) Maybe adjust the notice. "Is not allowed" sounds security related, but it's just that we haven't implemented it. We could say "creating append-distributed shards placements on the coordinator is not supported, skipping coordinator", or just omit the notice.

Reply:
@marcocitus done.
I think it's better to mention that we are skipping coordinator so I didn't omit it.